src/linearoperators/LBFGS.jl

```diff
@@ -36,52 +36,38 @@ end
 
 #default constructor
 
-function LBFGS(domainType, dim_in, M::I) where {I <: Integer}
-	s_M = [blockzeros(domainType, dim_in) for i = 1:M]
-	y_M = [blockzeros(domainType, dim_in) for i = 1:M]
-	s = blockzeros(domainType, dim_in)
-	y = blockzeros(domainType, dim_in)
-	T = typeof(s)
-	R = typeof(domainType) <: Tuple  ? real(domainType[1]) : real(domainType)
+function LBFGS(x::T, M::I) where {T <: AbstractArray, I <: Integer}
+	s_M = [0 .*similar(x) for i = 1:M]
+	y_M = [0 .*similar(x) for i = 1:M]
+	s = 0 .*similar(x)
+	y = 0 .*similar(x)
```

Comment:
You can just use `zero(x)` in all these here

Reply:
`zero(x::AbstractArray)` is no longer in use since Julia 1.0

Reply:
It’s there, and I think it was introduced in 0.7: https://docs.julialang.org/en/v1/base/numbers/#Base.zero

Reply:
I think they haven't update the docs... In Julia 1.1:
```julia
julia> x = randn(3); zeros(x)
ERROR: MethodError: no method matching zeros(::Array{Float64,1})
Closest candidates are:
  zeros(::Union{Integer, AbstractUnitRange}...) at array.jl:448
  zeros(::Type{T}, ::Union{Integer, AbstractUnitRange}...) where T at array.jl:449
  zeros(::Tuple{Vararg{Union{Integer, AbstractUnitRange},N} where N}) at array.jl:450
  ...
Stacktrace:
 [1] top-level scope at none:0
```

Reply:
ah right! in the past it was `zeros` now it's `zero`, that's why I got confused!